crates/cow-amm/src/amm.rs

```diff
@@ -86,6 +90,26 @@ impl Amm {
         let raw_signature = signature.0.into_iter().skip(20).collect();
         let signature = Signature::Eip1271(raw_signature);
 
+        // Cow AMM pools can have specific requirements for the signature validity.
```

Comment:
nit: The underlying issue is not that specific cow amms might have special requirements but rather that the helper contract might produce invalid signatures at all.

Reply:
But it's still Cow AMM specific EIP1271 signature check implementation causing the invalidity right... I mean, the order is good and signature is good, it's only bad in context of using that signature/order on cow amm pool.

Anyway, your statement is true as well. I think git blame can be used here to give more context in not clear.

Reply:
> But it's still Cow AMM specific EIP1271 signature check implementation causing the invalidity right.

The meaning of the signature is entirely implementation specific. If it doesn't work for the only thing it's supposed to work it's not really a good signature is it? 😅 
The correct behavior would be that the helper contract refuses to create the signature in the first place because it's supposed to be aware of the implementation details of the cow amms it's supposed to work for.
It's not the case that there is a catch all implementation of the helper contract which happens to be incompatible with the balancer cow amm. The helper was specifically written for the balancer version so we should be able to expect that it works.
But it's also true that by now there is enough context on the matter to be found with `git blame`.